pkg/operator/ceph/object/objectstore.go

```diff
@@ -777,6 +812,15 @@ func enableRGWDashboard(context *Context) error {
 		}
 		logger.Info("done setting the dashboard api secret key")
 	}()
+
+	defer func() {
```

Comment:
This defer needs to go inside the goroutine, for example on line 805 at the start of it. If it's not in the goroutine, this defer could delete the file before the goroutine uses it.